includes/class-wc-cart.php

```diff
@@ -1738,9 +1738,23 @@ public function add_discount( $coupon_code ) {
 
 		// If its individual use then remove other coupons
 		if ( $the_coupon->get_individual_use() ) {
-			$this->applied_coupons = apply_filters( 'woocommerce_apply_individual_use_coupon', array(), $the_coupon, $this->applied_coupons );
+			$coupons_to_keep = apply_filters( 'woocommerce_apply_individual_use_coupon', array(), $the_coupon, $this->applied_coupons );
+
+			foreach ( $this->applied_coupons as $applied_coupon ) {
+				$keep_key = array_search( $applied_coupon, $coupons_to_keep );
```

Comment:
Each coupon you don't want to keep you call `$this->remove_coupon( $applied_coupon );`. So when this loop is over, $this->applied_coupons is already up to date? So why do you then call `$this->applied_coupons += $coupons_to_keep;`? Or are they new ones?